src/Generator/Generators/CSharp/CSharpTextTemplate.cs

```diff
@@ -34,11 +34,27 @@ public static class Helpers
                 "void", "partial", "yield", "where"
             };
 
+        private static int numSafeIden = 0;
```

Comment:
I paid a little more attention to this now. I think it may cause problems. It is static which means that it will keep increasing for the generation of the entire library, even if the anonymous types reside in different name-spaces. I think we'd better check if the parent name-space already has a type of the same name and then modify it.